archivist_samples/document/document.py

```diff
@@ -0,0 +1,176 @@
+# pylint:disable=missing-function-docstring      # docstrings
+# pylint:disable=missing-module-docstring      # docstrings
+# pylint:disable=missing-class-docstring      # docstrings
+
+from importlib import resources
+
+import logging
+
+from copy import copy
+from typing import Dict, List, Optional
+
+# pylint:disable=unused-import      # To prevent cyclical import errors forward referencing is used
+# pylint:disable=cyclic-import      # but pylint doesn't understand this feature
+
+from archivist import archivist as type_helper
```

Comment:
After doing type-hinting for RF, I've come down to a particular style.  

```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
     from archivist.archivisti mport Archivist

```
Then use the type hints in quotation marks,  

```python
arch: "Archivist",
```
Additionally, you no longer need a majority of the typing imports. 
```python
Optional[list] -> "list|None"
Union[list, str] -> "list|str"
List[str] -> "list[str]"
```

Reply:
Excellent summary! have amended document.py to follow this pattern for type hinting.